### DISCUSSION TITLE
javaCard applet erreur 6F 00

### DISCUSSION THREAD

**Discussion by Racdia** [martinpaljak/GlobalPlatformPro#373] [Q&A]
Bonjour j'ai une erreur 6F 00 quand j'essaye de lire des informations sur ma carte a puce a travers mon applet  maintenant jessaye d'interoger la carte a travers l'invite de commande pour lire des données sur la carte mais quand j'envoi la requette suivant  voila ce que j'obtient  
gp.exe  --debug --apdu "00 A4 04 00 0A B4 F8 08 F1 85 00 00 02 00 00"
SCardConnect("Idemia Smartcard Reader 0", T=*) -> T=1, 3BDC18FF8191FE1FC38073C821136605036351000250
# GlobalPlatformPro 325fe84
# Running on Windows 10 10.0 amd64, Java 11.0.23 by Oracle Corporation
A>> T=1 (4+0010) 00A40400 0A B4F808F1850000020000
A<< (0000+2) (45ms) 6A82
A>> T=1 (4+0000) 00A40400 00
A<< (0018+2) (21ms) 6F108408A000000151000000A5049F6501FF 9000
Warning: no keys given, defaulting to 404142434445464748494A4B4C4D4E4F
SCardDisconnect("Idemia Smartcard Reader 0", true) tx:20/rx:22 
quand lance commande pour essayer de lire les informations sur la carte il me ramene 6A82  j'ai besoin d'aide

**Comment by martinpaljak** [comment]
This is English language forum.

**Reply by Racdia** [reply]
Hello I have an error 6F 00 when I try to read information on my smart card through my applet now I try to interrogate the card through the command prompt to read data on the card but when I send the following request this is what I get   
gp.exe --debug --apdu "00 A4 04 00 0A B4 F8 08 F1 85 00 00 02 00 00"
SCardConnect("Idemia Smartcard Reader 0", T=*) -> T=1, 3BDC18FF8191FE1FC38073C821136605036351000250
A>> T=1 (4+0010) 00A40400 0A B4F808F1850000020000
A<< (0000+2) (45ms) 6A82
A>> T=1 (4+0000) 00A40400 00
A<< (0018+2) (21ms) 6F108408A000000151000000A5049F6501FF 9000
Warning: no keys given, defaulting to 404142434445464748494A4B4C4D4E4F
SCardDisconnect("Idemia Smartcard Reader 0", true) tx:20/rx:22
when I run the command to try to read the information on the card it returns 6A82 I need help

**Reply by martinpaljak** [reply]
Where is the 6f00 ? 6a82 indicates "no such applet found". If you have issues _installing_ your applet, this is the right place. Other issues are better left for stackoverflow or similar.

**Reply by Racdia** [reply]
gp --list
Warning: no keys given, defaulting to 404142434445464748494A4B4C4D4E4F
ISD: A000000151000000 (INITIALIZED)
     Parent:   A000000151000000
     From:     A0000000620001
     Privs:    SecurityDomain, CardLock, CardTerminate, CardReset, CVMManagement, TrustedPath, AuthorizedManagement, TokenVerification, GlobalDelete, GlobalLock, GlobalRegistry, FinalApplication, ReceiptGeneration

APP: B4F808F18500 (SELECTABLE)
     Parent:   A000000151000000
     From:     B4F808F185CF

PKG: A0000001515350 (LOADED)
     Applet:   A000000151535041

PKG: D2760000986C6962 (LOADED)

PKG: D276000098636F7265 (LOADED)

PKG: D276000098657874 (LOADED)

PKG: D276000098636F72656B6F (LOADED)

PKG: D2760000984C4453 (LOADED)

PKG: D27600009854523033313130 (LOADED)

PKG: D276000098636F7265616F (LOADED)

PKG: D276000098636F726562696F (LOADED)

PKG: D276000098636F726572756C65 (LOADED)

PKG: D276000098454143 (LOADED)
     Applet:   A0000002471001

PKG: D276000098654944 (LOADED)
     Applet:   E80704007F00070302

PKG: D276000098455369676E (LOADED)
     Applet:   D27600009802

PKG: D276000098504B49 (LOADED)
     Applet:   A000000063504B43532D3135

PKG: D27600009865506173736C6574 (LOADED)
     Applet:   D27600009801

PKG: B4F808F185CF (LOADED)
     Applet:   B4F808F18500  

this the  list of applet on my javaCard my applet is B4F808F185CF but when i am trying to run this command  
>gp.exe  --debug --apdu "00 A4 04 00 0A B4 F8 08 F1 85 00 00 02 00 00"

i get this response SCardConnect("Idemia Smartcard Reader 0", T=*) -> T=1, 3BDC18FF8191FE1FC38073C821136605036351000250
# GlobalPlatformPro 325fe84
# Running on Windows 10 10.0 amd64, Java 11.0.23 by Oracle Corporation
A>> T=1 (4+0010) 00A40400 0A B4F808F1850000020000
A<< (0000+2) (47ms) 6A82
A>> T=1 (4+0000) 00A40400 00
A<< (0018+2) (27ms) 6F108408A000000151000000A5049F6501FF 9000
Warning: no keys given, defaulting to 404142434445464748494A4B4C4D4E4F
SCardDisconnect("Idemia Smartcard Reader 0", true) tx:20/rx:22

**Reply by martinpaljak** [reply]
Your app is (as listing tells):

APP: B4F808F18500 (SELECTABLE)

**Reply by Racdia** [reply]
yes that's my applet but when i'm trying send command   gp.exe --debug --apdu "00 A4 04 00 0A B4 F8 08 F1 85 CF 00 02 00 00"

i get this response  

SCardConnect("Idemia Smartcard Reader 0", T=*) -> T=1, 3BDC18FF8191FE1FC38073C821136605036351000250
# GlobalPlatformPro 325fe84
# Running on Windows 10 10.0 amd64, Java 11.0.23 by Oracle Corporation
A>> T=1 (4+0010) 00A40400 0A B4F808F185CF00020000
A<< (0000+2) (56ms) 6A82
A>> T=1 (4+0000) 00A40400 00
A<< (0018+2) (31ms) 6F108408A000000151000000A5049F6501FF 9000
Warning: no keys given, defaulting to 404142434445464748494A4B4C4D4E4F
SCardDisconnect("Idemia Smartcard Reader 0", true) tx:20/rx:22

can you explain me please this response ??

**Reply by Racdia** [reply]
gp -c B4F808F18500
[WARN] GPSession - No FCI returned to SELECT
Warning: no keys given, defaulting to 404142434445464748494A4B4C4D4E4F
Failed to open secure channel: INITIALIZE UPDATE failed: 0x6E00 (Invalid CLAss)
Read more from https://github.com/martinpaljak/GlobalPlatformPro/wiki/Keys

**Reply by Racdia** [reply]
hello
